packages/material-ui-styles/src/makeStyles.js

```diff
@@ -199,55 +191,52 @@ function makeStyles(stylesOrCreator, options = {}) {
       ...stylesOptions2,
     };
 
-    const { current: state } = React.useRef({});
+    const { current: instance } = React.useRef({
+      // previous: null,
+      // current: null,
+    });
+    const firstRender = React.useRef();
 
+    // ⚠️ You may rely on React.useMemo as a performance optimization, not as a semantic guarantee.
+    // https://reactjs.org/docs/hooks-reference.html#usememo
+    //
     // Execute synchronously every time the theme changes.
     React.useMemo(() => {
-      attach({
+      instance.current = {
```

Comment:
No side effects in render. Ever write to a ref belongs into an effect.

Reply:
I don't see any other possible alternative. How would you do it?